### DISCUSSION TITLE
Improve flows

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#15134] [Feature Requests]
Hi,
I am a very deep user as flows, some ideas of flows:

1. split success log and fail log(any step rejected), then we are easy to track system exceptions.

2. automatically clean up flows log.

3. add a search component on the frontend, because I have over 100 flows.

4. flows log searchable.

5. flows frontend page add flow_type field (event hook | manually | schedule )

6. make UpdateData operation filter field is required, because of if we forgot set the filter. will update a lot of records, and support updateByPK / QueryByPK.

**Comment by benhaynes** [comment]
What about an included flow recipe (disabled by default)... to prune the flow logs?! 😉

But yes, a built in option for this would be nice.

**Reply by imiskolee** [reply]
i think having a built-in option is better. and can be set the log lifetime, default as 0 days(keep log forever).

**Comment by erondpowell** [comment]
@imiskolee

Glad to see you're testing the frontiers with Flows! 
I'm rewriting/restructuring the documentation to make it more robust.

If you have any docs/learning/debugging related feedback you'd like to share, I'd love to hear 🙏 
If interested, we could correspond here or hop on a discord video chat!

谢谢！

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
